src/ts/forwarders/proxyaccountfowarder.ts

```diff
@@ -0,0 +1,192 @@
+import { keccak256, defaultAbiCoder, solidityKeccak256 } from "ethers/utils";
+import { Wallet } from "ethers/wallet";
+import { ReplayProtectionAuthority } from "../replayprotection/replayprotectionauthority";
+import { ChainID, ProxyAccountDeployer, ProxyAccountFactory } from "../..";
+import {
+  ForwardParams,
+  Forwarder,
+  ProxyCallData,
+  DeploymentParams,
+} from "./forwarder";
+import { Create2Options, getCreate2Address } from "ethers/utils/address";
+
+export interface CreateProxyData {
+  to: string;
+  callData: string;
+}
+/**
+ * A single library for approving meta-transactions and its associated
+ * replay protection. All meta-transactions are sent via proxy contracts.
+ */
+export class ProxyAccountForwarder extends Forwarder<ProxyCallData> {
+  private baseAccount: string;
+  /**
+   * All meta-transactions are sent via an proxy contract.
+   * @param chainID Chain ID
+   * @param proxyDeployer Address of contract
+   * @param signer Signer's wallet
+   * @param replayProtectionAuthority Extends implementation ReplayProtectionAuthority
+   */
+  constructor(
+    chainID: ChainID,
+    private readonly proxyDeployer: ProxyAccountDeployer,
+    signer: Wallet,
+    replayProtectionAuthority: ReplayProtectionAuthority
+  ) {
+    super(chainID, signer, replayProtectionAuthority);
+  }
+
+  /**
+   * Standard encoding for contract call data
+   * @param data The target contract, value (wei) to send, and the calldata to execute in the target contract
+   */
+  protected getEncodedCallData(data: ProxyCallData) {
+    // ProxyAccounts have a "value" field.
+    return defaultAbiCoder.encode(
+      ["address", "uint", "bytes"],
+      [data.target, data.value, data.callData]
+    );
+  }
+
+  /**
+   * Checks if the ProxyContract is already deployed.
+   * @returns TRUE if deployed, FALSE if not deployed.
+   */
+  public async isProxyContractDeployed(): Promise<boolean> {
+    const deployed = await this.proxyDeployer
+      .connect(this.signer)
+      .accounts(this.signer.address);
+
+    // Does the user have a proxy contract?
+    if (deployed === "0x0000000000000000000000000000000000000000") {
+      return false;
+    }
+
+    return true;
+  }
+  /**
+   * Returns the encoded calldata for creating a proxy contract
+   * No need for ForwardParams as no signature is required in ProxyAccountDeployer
+   * @returns The proxy deployer address and the calldata for creating proxy account
+   * @throws If the proxy account already exists
+   */
+  public async createProxyContract(): Promise<CreateProxyData> {
```

Comment:
Should probably return the amount of gas required as well no?

Reply:
requires about ~115k gas inc transaction. i've added a gas variable and a unit test